server/routerlicious/packages/test-utils/src/testCache.ts

```diff
@@ -0,0 +1,16 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { ICache } from "@fluidframework/server-services-core";
+
+export class TestCache implements ICache {
+    private readonly map = new Map<string, string>();
+    public async get(key: string): Promise<string> {
+        return this.map.get(key) ?? "";
```

Comment:
I think Redis get() returns null when the key does not exist (https://redis.io/commands/get). Does that value become empty string in ioredis, so you are mimicking that behavior here? In line 152 of auth.ts, would the if condition evaluate to false if the key does not exist and we return `""` in the TestCache?

Reply:
Good point, I guess the TestCache might not be a perfect representation of the Redis cache. However, I think the behavior is the same.

Condition for forbidding access:
- JWT has already been used and is cached. Line 152 in auth.ts returns `"used"`

Conditions for allowing access:
- JWT has not been used and is not cached. Line 152 in auth.ts returns `null`, `""`, `undefined`
- Redis (or other caching mechanism) fails unexpectedly, throwing an error. Line 152 in auth.ts returns `false` thanks to catch statement
  - This is the questionable part. Should we block document creation when Redis encounters an error? If it is transient, we should still block _most_ malicious document creation spam, which is the purpose of this change. If it is permanent/long-lasting, many other things will be failing, but documents would still be created which could be an issue

Reply:
I agree that the question above is something for us to discuss, but what I meant above in this comment is whether returning `""` in line 11 above would work with the logic on auth.ts line 152. The reason I say that is because
`this.map.get(key)` would return undefined if the key is not there. If that is the case, we would like auth.ts to accept the request. But in line 11, if `this.map.get()` is undefined, we are returning "". Does empty string evaluate to true or false? (dumb question). If it evaluates to false, then no problem, since the if statement in auth.ts line 152 would not be executed and we would not reject the request, as expected, since the key was not in the test map.

Reply:
> Does empty string evaluate to true or false?

it evaluates to ["falsy"](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)

Reply:
Then looks like we are all good :)